### PR TITLE
Feature/android player

### PR DESCRIPTION
**Pull request by arokor**
Add Android player MVP

Lots of generated files in this PR.
There is more to do here but this is a good foundation to build from:
- Embeds player code
- Implements machine-  and cache interfaces
- Some working unit tests for typescript code (for the interface implementations mentioned above)
- A capacitor plugin that lets us call native java code from typescript
- A working unit test that test the java implementation of this plugin.
- App starts on boot
- Interacts with separate watchdog app when available

### CODE REVIEW

File: packages/castmill/lib/castmill/workers/image_transcoder.ex [hunk 1]
```diff
@@ -52,7 +52,7 @@ defmodule Castmill.Workers.ImageTranscoder do
           {uri, size} =
             Image.open!(stream)
             |> Image.thumbnail!(size)
-            |> upload_image(organization_id, media_id, "#{context}.#{extension}")
+            |> upload_image(organization_id, media_id, "#{context}#{extension}")
```
Comment (manast) [inline packages/castmill/lib/castmill/workers/image_transcoder.ex:55]: Is this correct? we need the period to get a file extension, like "preview.jpg", otherwise we would get "previewjpg"

File: packages/castmill/lib/castmill_web/endpoint.ex [hunk 2]
```diff
@@ -83,6 +83,7 @@ defmodule CastmillWeb.Endpoint do
   # TODO: we must cache the allowed origins at least for a few minutes or this
   # will be a huge performance bottleneck. The cache must use ETS to be shared.
   def getAllowedOrigins() do
-    Castmill.Networks.list_network_domains()
+    ["https://localhost" | Castmill.Networks.list_network_domains()]
```
Comment (manast) [inline packages/castmill/lib/castmill_web/endpoint.ex:86]: Is this debug code? I see an https protocol which is difficult to replicate locally, but in any case list_network_domains() should return the allowed domains correctly.

File: packages/device/src/classes/device.ts [hunk 3]
```diff
@@ -338,13 +338,13 @@ export class Device extends EventEmitter {
   }
 
   get socketEndpoint() {
-    return `${this.opts?.baseUrl || ''}/socket`;
+    return `${this.getBaseUrl().replace('http', 'ws')}/socket`;
```
Comment (manast) [inline packages/device/src/classes/device.ts:341]: what about https and wss ?

File: packages/platforms/android-player/android/app/src/androidTest/java/com/getcapacitor/myapp/ExampleInstrumentedTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.getcapacitor.myapp;
```
Comment (manast) [inline packages/platforms/android-player/android/app/src/androidTest/java/com/getcapacitor/myapp/ExampleInstrumentedTest.java:1]: Is this file needed? (androidTest, etc)

File: packages/platforms/android-player/src/index.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    <meta
+      name="viewport"
+      content="viewport-fit=cover, width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=1.0, user-scalable=no"
+    />
+    <meta name="format-detection" content="telephone=no" />
+    <meta name="msapplication-tap-highlight" content="no" />
+
+    <script
+      type="module"
+      src="https://unpkg.com/@ionic/pwa-elements@latest/dist/ionicpwaelements/ionicpwaelements.esm.js"
```
Comment (manast) [inline packages/platforms/android-player/src/index.html:15]: wouldn't this be a problem if the player starts offline?

File: packages/platforms/android-player/src/ts/app.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+const a: number = 1;
+
+console.log('Hello Castmill', a);
```
Comment (manast) [inline packages/platforms/android-player/src/ts/app.ts:3]: What is this file needed for?

File: packages/platforms/android-player/src/ts/components/player-frame.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+      return;
+    }
+
+
+    const androidMachine = new AndroidMachine();
+    const cache = new AndroidStorage('file-cache');
+
+    // Android Emulator uses 10.0.2.2 to refer to the host machine
+    // TODO: Make this configurable somehow to support different player backends
+    const device = new Device(androidMachine, cache, {baseUrl: 'http://10.0.2.2:4000'});
```
Comment (manast) [inline packages/platforms/android-player/src/ts/components/player-frame.tsx:19]: where does 10.0.2.2 come from?

**Review by manast — APPROVED** [review]
LGTM, I wrote some small comments though.

File: packages/castmill/lib/castmill/workers/image_transcoder.ex (hunk 1 above)
Comment (arokor) [reply packages/castmill/lib/castmill/workers/image_transcoder.ex:55]: extension already includes the `"."` I noticed that we had files like "preview..jpg" and changed this to get rid of the double dots.

File: packages/castmill/lib/castmill_web/endpoint.ex (hunk 2 above)
Comment (arokor) [reply packages/castmill/lib/castmill_web/endpoint.ex:86]: It is needed to make it work on android where the local files are served from https://localhost

File: packages/device/src/classes/device.ts (hunk 3 above)
Comment (arokor) [reply packages/device/src/classes/device.ts:341]: Should work for https as well
```js
'http://example.com'.replace('http', 'ws') === 'ws://example.com'
'https://example.com'.replace('http', 'ws') === 'wss://example.com'
```

File: packages/platforms/android-player/src/index.html (hunk 5 above)
Comment (arokor) [reply packages/platforms/android-player/src/index.html:15]: Generated by capacitor. I don't think they are used. Will remove

File: packages/platforms/android-player/src/ts/components/player-frame.tsx (hunk 7 above)
Comment (arokor) [reply packages/platforms/android-player/src/ts/components/player-frame.tsx:19]: It's the IP of the host machine when running in Android emulator. I'll make this configurable in an upcoming PR

File: packages/platforms/android-player/android/app/src/androidTest/java/com/getcapacitor/myapp/ExampleInstrumentedTest.java (hunk 4 above)
Comment (arokor) [reply packages/platforms/android-player/android/app/src/androidTest/java/com/getcapacitor/myapp/ExampleInstrumentedTest.java:1]: No. It was generated by capacitor and I'll remove it

File: packages/platforms/android-player/src/ts/app.ts (hunk 6 above)
Comment (arokor) [reply packages/platforms/android-player/src/ts/app.ts]: Nothing at all... Removed.

File: packages/platforms/android-player/src/index.html (hunk 5 above)
Comment (arokor) [reply packages/platforms/android-player/src/index.html]: removed
